### ISSUE TITLE
Configuration fails with Xcode new build system

### ISSUE THREAD

**Issue by szhorvat** [igraph/igraph#2141]
**Describe the bug**

Configuring igraph fails when using the Xcode new build system. This should be fixed as using Xcode is by far the easiest way to run the Clang static analyser and get a usable report.

Assigning this to you @ntamas as you are the only one with a permanent access to recent macOS versions.

**To reproduce**

Use a recent version of Xcode (which requires a recent version of macOS — so not 10.14).

Run `cmake .. -GXcode`.

```
CMake Error in src/CMakeLists.txt:
  The custom command generating

    /Users/szhorvat/Repos/igraph-main/igraph/build/src/io/parsers/dl-parser.c

  is attached to multiple targets:

    parsersources
    igraph

  but none of these is a common dependency of the other(s).  This is not
  allowed by the Xcode "new build system".
```

**Version information**

Both `master` and `develop`

**Comment by szhorvat** [comment]
This may work, but can't test it now:

```cmake
if (NOT CMAKE_XCODE_BUILD_SYSTEM EQUAL 12)
  add_custom_target(parsersources SOURCES ${PARSER_SOURCES})
endif()
```

**Comment by ntamas** [comment]
There are still some errors elsewhere:

```
CMake Error in doc/CMakeLists.txt:
  The custom command generating

    /Users/tamas/dev/igraph/igraph/build-xcode/doc/arpack.xml

  is attached to multiple targets:

    html
    pdf

  but none of these is a common dependency of the other(s).  This is not
  allowed by the Xcode "new build system".
```

I'll take care of these.

**Comment by ntamas** [comment]
Should be fixed in 60cbccf81
